app/src/main/java/net/aiscope/gdd_app/ui/metadata/SampleImagesAdapter.kt

```diff
@@ -86,73 +89,47 @@ private class ImageViewHolder(
         itemView.setOnClickListener { onImageClicked(imageFile, maskFile) }
     }
 
-    fun bind(image: File, mask: File) {
+    fun bind(image: File, mask: File, cacheDir: File) {
         imageFile = image
         maskFile = mask
 
         (itemView.tag as? Job)?.cancel()
         itemView.tag = uiScope.launch {
             (itemView as ImageView).setImageBitmap(null)
-            val bitmap = decodeSampledBitmapFromResource(
+            val bitmap = decodeSampledBitmapAndCache(
                 image,
                 itemView.context.resources.getDimensionPixelSize(R.dimen.sample_image_thumbnail_width),
-                itemView.context.resources.getDimensionPixelSize(R.dimen.sample_image_thumbnail_height)
+                itemView.context.resources.getDimensionPixelSize(R.dimen.sample_image_thumbnail_height),
+                itemView.context.cacheDir
             )
             itemView.setImageBitmap(bitmap)
         }
     }
 }
 
-suspend fun decodeSampledBitmapFromResource(
+suspend fun decodeSampledBitmapAndCache(
     image: File,
     reqWidth: Int,
-    reqHeight: Int
+    reqHeight: Int,
+    cacheDir: File
 ): Bitmap = withContext(Dispatchers.IO) {
     val cachedImage = File(
-        image.parent,
+        cacheDir,
         "${image.nameWithoutExtension}_${reqWidth}x${reqHeight}.${image.extension}"
     )
     if (cachedImage.exists()) {
         return@withContext BitmapFactory.decodeFile(cachedImage.absolutePath)
     }
-    // First decode with inJustDecodeBounds=true to check dimensions
-    BitmapFactory.Options().run {
-        inJustDecodeBounds = true
-        BitmapFactory.decodeFile(image.absolutePath, this)
-
-        // Calculate inSampleSize
-        inSampleSize = calculateInSampleSize(this, reqWidth, reqHeight)
-
-        // Decode bitmap with inSampleSize set
-        inJustDecodeBounds = false
-
-        val bitmap = BitmapFactory.decodeFile(image.absolutePath, this)
+    val bitmap = BitmapReader.decodeSampledBitmapFromResource(
+        image,
+        false,
```

Comment:
Would you use a named argument here, so it's clear what does the `false` stands for, please?

Reply:
Had to change the order since it's not allowed to mix positional & named arguments, but done